### PR TITLE
Supports task-specific sample weights, weighed_metrics and fixes InputBlockV2

### PR DESCRIPTION
**Pull request by gabrielspmoreira**
### Context
Currently, if `sample_weight` is provided it is used for computing the loss and metrics for all prediction tasks.
Although, for Multi-Task Learning, it is sometimes necessary to use different `sample_loss` for different prediction tasks.

For example, when training a ranking model that predicts both a click or a purchase we have two separate binary classification tasks. Those the two task losses are summed automatically, with weights optionally defined in `model.compile(loss_weights=)`.

But you might want to have different sample weights for those different tasks. For example, in this use case, `purchase==1`  is only possible if `clicked==1`. So the model gets better results if the loss for purchase target is computed only for clicked examples, to make the purchase target less sparser, like proposed in the [PLE paper](https://dl.acm.org/doi/10.1145/3383313.3412236) about Multi-Task Learning. That can be implemented by giving `sample_weight=0` for purchase task when `clicked==0`, which is what we allow with this PR.

### Goals :soccer:
- [x] Enable support for task-specific sample weights
- [x] Enable support for task-specific weighted metrics
- [x] Adds examples on how to set `loss_weights`, `sample_weights` and `weighted_metrics` in `model.compile()`
- [x] Fixes bugs of `InputBlockV2` and `EmbeddingTable` in graph mode

### Implementation Details :construction:
- In this PR we make it possible to set task-specific `sample_weights` and also to manipulate any information in `PredictionOutput`, like `predictions`, `targets`, `masks`.
- The `sample_weight` can be provided by mapping the dataloader to return a triple (features, targets, sample_weight), or for task-specific `sample_weight` they can be set by means of a `pre` block for the `PredictionTask`
- The `sample_weight` is used to compute the loss and also the `weighted_metrics`. The  `metrics` are not affect by `sample_weight`

### Testing Details :mag:
The new test `test_mmoe_head_task_specific_sample_weight` demonstrates how it is possible to create a custom block that sets a task-specific `sample_weight` based on a feature available in `features` or `targets`.
I also included tests to demonstrate how `loss_weights` can be used to set task-specific weights for the final loss.

### CODE REVIEW

File: tests/unit/tf/prediction_tasks/test_multi_task.py [hunk 1]
```diff
@@ -75,6 +86,91 @@ def test_mmoe_head(music_streaming_data: Dataset):
...
+            tf.keras.metrics.AUC(name="weighted_auc"),
+        ),
+        "play_percentage/regression_task": (
+            tf.keras.metrics.RootMeanSquaredError(name="weighted_root_mean_squared_error"),
+        ),
+    }
+
+    model.compile(
+        optimizer="adam",
+        run_eagerly=True,
```
[the remark this answers is not in the document]
Comment (gabrielspmoreira) [reply tests/unit/tf/prediction_tasks/test_multi_task.py]: Not a specific reason for that. Changed the multi_task tests to run in both eager and graph mode
